pydivide/utilities.py

```diff
@@ -493,23 +493,37 @@ def get_latest_files_from_date_range(date1, date2):
         month = str('%02d' % current_date.month)
         day = str('%02d' % current_date.day)
         full_path = os.path.join(maven_data_dir, year, month)
-        if os.path.exists(full_path): 
-            version = 0
-            revision = 0
+        if os.path.exists(full_path):
+            versions = []
+            revisions = []
+            crustal = []
+            i = 0
             for f in os.listdir(full_path):
                 if kp_regex.match(f).group('day') == day:
                     v = kp_regex.match(f).group('version')
-                    if int(v) > int(version):
+                    if int(v) > 0:
                         version = v
-            for f in os.listdir(full_path):
-                if kp_regex.match(f).group('day') == day and kp_regex.match(f).group('version') == version:
+                        versions.append(version)
                     r = kp_regex.match(f).group('revision')
-                    if int(r) > int(revision):
+                    if int(r) > 0:
                         revision = r
-            if int(version) > 0:
-                seq = ('mvn', 'kp', 'insitu', year + month + day, 'v' + str(version), 'r' + str(revision) + '.tab')
-                filenames.append(os.path.join(full_path, '_'.join(seq)))
-                
+                        revisions.append(revision)
+                if kp_regex.match(f).group('day') == day and kp_regex.match(f).group('description') == '_crustal':
+                    crustal.append(True)
+                if kp_regex.match(f).group('day') == day and not kp_regex.match(f).group('description'):
+                    crustal.append(False)
+                i += 1
+            for v, ver in enumerate(versions):
```

Comment:
It looks like this function now just returns all versions and revisions of a file, but before it would only return the latest version/revisions.  The pydivide.read() function reads in all files returned by this function, so it would read in every version and revision of the file that the user has on their hard drive.